### ISSUE TITLE
Resize VST window

### ISSUE THREAD

**Issue by cocreature** [zynaddsubfx/zyn-fusion-issues#57]
Currently I am unable to resize the VST window in both Carla and Renoise. I vaguely recall talking to falktx quite some time ago and he mentioned some problems with doing that, but maybe that was specific to Carla. It looks like Renoise supports VST resizing at least on windows according to http://forum.renoise.com/index.php/topic/45386-resize-vst-windows/ but I couldn’t figure out whether it is also supported on Linux. https://forum.juce.com/t/a-few-fixes-for-linux-vst/5818 suggests so but that post is 6 years old. Since Renoise does not support LV2, I have to stick to VSTs even on Linux.

**Comment by mikobuntu** [comment]
I believe resizing is only available in standalone and LV2 versions at the moment. I'm pretty sure @fundamental plans to allow resizing of the VST version in the future.

**Comment by fundamental** [comment]
VST resizing is a bit odd compared to how resizing normally works.
When running the standalone you can either grab a corner of the window,
maximize it, or use other controls the window manager provides to
initiate a resize event.
When this happens the application (zyn-fusion in this case) is told
"someone wants you to resize to WxH" and then on the next drawing frame
it resizes to the new dimensions and renders the new graphics.
It works the same in the LV2 version as of 3.0.1.

VST on the other hand decided to make itself different in a way that I
consider a really dumb design decision on their part.
The plugin host doesn't tell the plugin to resize and it doesn't let the
window manager tell the plugin to resize either.
What it does allow is it allows for the plugin to issue resize events
for itself.
In the case of the planned implementation zyn-fusion will provide a list
of alternative sizes (and perhaps a custom size option) for the window
to resize to.
When the size is changed within this list the zyn-fusion will issue a
resize event which is then propagated to the plugin host which then
changes the size of the window, which then allows zyn-fusion to resize,
and then resizing proceeds as normal.
LV2/Standalone versions could also use this inverted resizing operation,
but I'm pretty sure most people want to just grab the corner of the app
and use that to resize the darn thing.

In summary:
- I need to add some control within the zyn-fusion UI to generate resize
  events
- I need to make sure that those events are propagated back to the
  zynaddasubfx VST code
- I should add the preferred size to the zynaddsubfx global config file
- This will likely take place when V1 of the settings view is written

**Comment by unfa** [comment]
I'm using Carla-git and Zyn-Fusion LV2 and can't resize the window either.
PS: Same in Ardour. I can resize the Ardour's plugin window, but Zyn-Fusion stays small within that big window.
